### PR TITLE
DSD-770: Table Row Header color bug fix

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-770](https://jira.nypl.org/browse/DSD-770)

## This PR does the following:

- Using `useRowHeaders` sets the first cell in each row as a `th`. This is as expected. The problem comes when custom header colors are added. If a dark background color and a light text color are passed, the `thead` header row displays correctly, but the first cell in each row in `tbody` does not. This is very apparent if the background color is black and the text color is white. It'll appear as if the row header cells are gone. This PR fixes that issue.
- Also adds more tests and updates the Storybook file with more examples.

## How has this been tested?

Locally and through Storybook.

## Accessibility concerns or updates

- N/A. 

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [x] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr872-sbi2fv7thtexcdaivkucdjakfcqsxia4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620c2075dae2dd15a59d5ccb

File: src/theme/components/customTable.ts [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
 const baseStyle = ({
   columnHeadersTextColor,
   showRowDividers,
   useRowHeaders,
 }) => ({
+  // Just so we don't let row headers to display as white. Only the
+  // thead should have white text color with any color background.
+  tbody: {
+    th: {
+      color:
```
Comment (gonuguntla) [inline src/theme/components/customTable.ts:20]: Excellent

File: src/theme/components/customTable.ts [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+        useRowHeaders && whiteColors.includes(columnHeadersTextColor)
+          ? "var(--nypl-colors-ui-black)"
+          : columnHeadersTextColor,
+    },
+  },
```
Comment (gonuguntla) [inline src/theme/components/customTable.ts:14]: Screen Shot 2022-02-15 at 5 12 53 PM

Comment (gonuguntla) [reply src/theme/components/customTable.ts:14]: @EdwinGuzman  I added **columnHeadersBackgroundColor** also "#FFF".   I am thinking  that both colors should not match.

Comment (EdwinGuzman) [reply src/theme/components/customTable.ts:14]: Hmmm that's a good point. I don't think anyone will purposely add white to both props... The initial issue that is fixed with this PR can happen by accident so it's more important. We can set this aside until we implement a color contrast checker or a better color validation repo-wide. I'm saying this as this issue also happens in other components such as the `Card`:

Screen Shot 2022-02-15 at 6 02 36 PM

So we can fix all of this in another PR.

Comment (gonuguntla) [reply src/theme/components/customTable.ts:14]: Ok Edwin

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr872-sbi2fv7thtexcdaivkucdjakfcqsxia4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620c2075dae2dd15a59d5ccb

File: src/theme/components/customTable.ts (hunk 1 above)
Comment (bigfishdesign13) [inline src/theme/components/customTable.ts]: I misunderstood how this was going to be addressed.  The row headers should simply stay black all the time and we should only allow the color of the column headers to be changed.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Sorry, I misunderstood how you were going to address the text color issue.  Please see my comment.  

Also, I appear to have overlooked the styles applied to the `caption` element, so let’s clean those up.  Please add the following to the styles for  `caption`:

```
padding: “0”,
marginBottom: “s”,
marginLeft: “0”,
marginRight: “0”,
marginTop: “0”,
```

And remove `textTransform: “capitalize”,` from `caption`.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr872-sbi2fv7thtexcdaivkucdjakfcqsxia4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620c2075dae2dd15a59d5ccb

File: src/theme/components/customTable.ts [hunk 3]
```diff
@@ -35,11 +44,13 @@ const baseStyle = ({
   },
   caption: {
     captionSide: "top",
-    textAlign: "left",
-    fontSize: "heading.secondary",
     color: "ui.black",
-    textTransform: "capitalize",
+    fontSize: "heading.secondary",
     fontWeight: "heading.secondary",
+    margin: "0",
```
Comment (bigfishdesign13) [inline src/theme/components/customTable.ts]: Do you think that CSS issue would be resolved if each margin direction was set separately?

**Review by bigfishdesign13 — APPROVED** [review]
Works well.

File: src/theme/components/customTable.ts (hunk 3 above)
Comment (EdwinGuzman) [reply src/theme/components/customTable.ts]: Doesn't hurt to do.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr872-sbi2fv7thtexcdaivkucdjakfcqsxia4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620c2075dae2dd15a59d5ccb

**Pull request merged**